src/main/java/job/dispatcher/MemoryDispatcher.java

```diff
@@ -0,0 +1,81 @@
+package job.dispatcher;
+
+import job.dispatcher.http.HttpRequest;
+import lombok.Getter;
+import lombok.Synchronized;
+import models.JobContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.LinkedList;
+
+@Getter
+public class MemoryDispatcher implements IDispatcher{
+    //private AbstractDataStore abstractDataStore;
+    private HttpRequest httpRequest;
+    private LinkedList<JobContext> jobContextList;
+    private Logger log = LoggerFactory.getLogger(MemoryDispatcher.class.getName());
+
+    @Inject
+    public MemoryDispatcher(LinkedList<JobContext> jobContextList, HttpRequest httpRequest){
+        this.jobContextList = jobContextList;
+        this.httpRequest = httpRequest;
+    }
+
+    public LinkedList<JobContext> getJobContextList(){
+        return jobContextList;
+    }
+
+    public void insert(JobContext newJob) {
+        log.info("Inserting job {} in the dispatcher", newJob.getJobId());
+
+        if(jobContextList.isEmpty()){
+            jobContextList.addFirst(newJob);
+            return;
+        }
+
+        int index = 0;
+        JobContext start = jobContextList.get(index);
+
+        if(start.getNextScheduleEpoch() > newJob.getNextScheduleEpoch()){
```

Comment:
Isn't it better to have a PriorityQueue where jobs are prioritised based on scheduled epochs?